rust/src/delta.rs

```diff
@@ -561,6 +569,34 @@ impl DeltaTable {
         }
     }
 
+    pub fn get_files_by_partitions(
+        &self,
+        filters: Vec<PartitionFilter<&str>>,
+    ) -> Result<Vec<String>, DeltaTableError> {
+        let partitions_number = match &self.state.current_metadata {
+            Some(metadata) if !metadata.partition_columns.is_empty() => {
+                metadata.partition_columns.len()
+            }
+            _ => return Err(DeltaTableError::LoadPartitions),
```

Comment:
the error message thrown by this error is not accurate, because this match branch could be reached when table struct is created, but having loaded any transaction log yet.

perhaps a better way to deal with this is to invoke `self.get_metadata()?` .

Reply:
I agree! I will separate the error handling into two separate calls:

1. No metadata error 
2. No partition column found (feel free to tell me if the error message is relevant)